File: src/main/java/study/java/spring/collection/TargetItem.java
package study.java.spring.collection;

public class TargetItem implements Comparable<TargetItem> {
    private final long id;
    private final String name;
    private final int priority;

    public TargetItem(long id, String name, int priority) {
        this.id = id;
        this.name = name;
        this.priority = priority;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getPriority() {
        return priority;
    }

    @Override
    public int compareTo(TargetItem o) {
        int priorityCompareResult = Integer.compare(priority, o.priority);
        return priorityCompareResult != 0 ? priorityCompareResult : Long.compare(id, o.id);
    }
}

File: src/main/java/study/java/spring/collection/TargetItem.kt
package study.java.spring.collection

class TargetItem(val id: Long, val name: String, val priority: Int): Comparable<TargetItem> {
    override fun compareTo(o: TargetItem): Int {
        val priorityCompareResult = priority.compareTo(o.priority)
        return if (priorityCompareResult != 0) priorityCompareResult else id.compareTo(o.id)
    }
}
